code/datums/action.dm

```diff
@@ -65,6 +67,12 @@
 		return FALSE
 	return TRUE
 
+/datum/action/proc/AltTrigger()
+	if(!has_altclick)
+		Trigger()
+		return FALSE
+	return TRUE
```

Comment:
why not make this just call Trigger() and return FALSE then remove the `has_altclick` var because if it has an alt click the child proc will override this so theres no need to store `has_altclick` as a variable